storage/operation/children_test.go

```diff
@@ -20,27 +21,25 @@ func TestBlockChildrenIndexUpdateLookup(t *testing.T) {
 		childrenIDs := unittest.IdentifierListFixture(8)
 		var retrievedIDs flow.IdentifierList
 
-		lctx := lockManager.NewContext()
-		err := lctx.AcquireLock(storage.LockInsertBlock)
-		require.NoError(t, err)
-
-		err = db.WithReaderBatchWriter(func(rw storage.ReaderBatchWriter) error {
-			return operation.UpsertBlockChildren(lctx, rw.Writer(), blockID, childrenIDs)
+		unittest.WithLock(t, lockManager, storage.LockInsertBlock, func(lctx lockctx.Context) error {
+			err := db.WithReaderBatchWriter(func(rw storage.ReaderBatchWriter) error {
```

Comment:
delegate error handling (or pass through, if you happen to go with my suggestion) to `WithLock`